ovs/lib/storagerouter.py

```diff
@@ -108,22 +108,14 @@ def get_metadata(storagerouter_guid):
                 used_space_by_roles = 0
                 for storagedriver_partition in disk_partition.storagedrivers:
                     claimed_space += storagedriver_partition.size if storagedriver_partition.size is not None else 0
-                    _directory_used_size = 0
-                    try:
-                        output = client.run('du -Pl {0}'.format(storagedriver_partition.path))
-                    except Exception as ex:
-                        logger.error('Failed to get directory usage for {0}. {1}'.format(storagedriver_partition.path, ex))
-                    else:
+                    directory_used_size = 0
+                    if client.dir_exists(storagedriver_partition.path):
                         try:
-                            _used_size, _ = output.split('\t')
-                        except ValueError as ve:
-                            logger.error('Unexpected output for directory {0}. "{1}" . {2}'.format(storagedriver_partition.path, output, ve))
-                        else:
-                            try:
-                                _directory_used_size = int(_used_size) * 1024
-                            except ValueError as ve:
-                                logger.error('Could not parse value: {0}. {1}'.format(_used_size, ve))
-                    used_space_by_roles += _directory_used_size
+                            used_size, _ = client.run('du -B 1M -d 0 {0}'.format(storagedriver_partition.path)).split('\t')
+                            directory_used_size = int(used_size) * 1024
```

Comment:
You still do the multiplication?
Using the -B 1M should already return the value in megabytes and this is what you need right?